lib/registration.pm

```diff
@@ -190,7 +190,14 @@ sub fill_in_registration_data {
     if (sle_version_at_least('15') && check_var('DISTRI', 'sle')) {
         my $modules_needle = "modules-preselected-" . get_required_var('SLE_PRODUCT');
         if (get_var('UPGRADE') || get_var('PATCH')) {
-            check_screen($modules_needle, 5);
+            # Make sure patch_before_migration is finished and in upgrading process
+            if (check_var('HDDVERSION', get_var('VERSION'))) {
+                # Workaround for bsc#1070031: Skip "Extension and Module Selection" in upgrading
+                unless (check_screen($modules_needle, 5)) {
+                    record_soft_failure('bsc#1070031');
+                    return;
+                }
+            }
```

Comment:
That's another two level of indentation. I recommend to extract that block into a function for better readability

Reply:
Hmm, actually there is potential issue that I didn't consider upgrading without var PATCH=1.